src/main/java/bitflow4j/io/file/OutputFiles.java

```diff
@@ -0,0 +1,118 @@
+package bitflow4j.io.file;
+
+import bitflow4j.AbstractPipelineStep;
+import bitflow4j.Header;
+import bitflow4j.Sample;
+import bitflow4j.io.MarshallingSampleWriter;
+import bitflow4j.io.marshall.Marshaller;
+import bitflow4j.script.registry.Description;
+
+import java.io.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Created by kevinstyp on 15/04/2019.
+ */
+@Description("Uses the provided template in parameter file to save samples in their respective file. Example call: " +
+        "output-files('./${data_type}/${serial}.csv', 'CSV') " +
+        "This will evaluate the tags data_type and serial of every sample and save the sample in the proper file.")
+public class OutputFiles extends AbstractPipelineStep {
+
+    private static final Logger logger = Logger.getLogger(OutputFiles.class.getName());
+
+    private final Marshaller marshaller;
+    protected OutputStream output = null;
+    private final Header.ChangeChecker header = new Header.ChangeChecker();
+
+    private final boolean append;
+    private int index = 0;
+    private String baseFileTemplate;
+
+
+    public OutputFiles(String file, String marshaller) throws IOException {
```

Comment:
To be compatible with other implementations, it would be good to have a constructor without the separate marshaller argument. It can be determined automatically using the file ending... Default should be CSV, while ".bin" results in the binary marshaller. Please check, if there is a similar functionality in the FileSink or some related class.